fv3core/stencils/fvtp2d.py

```diff
@@ -82,10 +83,21 @@ def __init__(self, namelist, hord):
         self.stencil_runtime_args = {
             "validate_args": global_config.get_validate_args(),
         }
-        stencil_wrapper = gtscript.stencil(**stencil_kwargs)
-        self.stencil_q_i = stencil_wrapper(q_i_stencil)
-        self.stencil_q_j = stencil_wrapper(q_j_stencil)
-        self.stencil_transport_flux = stencil_wrapper(transport_flux_xy)
+        self.stencil_q_i = FixedOriginStencil(
+            q_i_stencil,
+            origin=self.grid.full_origin(add=(0, 3, 0)),
+            domain=self.grid.domain_shape_full(add=(0, -3, 1)),
+        )
+        self.stencil_q_j = FixedOriginStencil(
+            q_j_stencil,
+            origin=self.grid.full_origin(add=(3, 0, 0)),
+            domain=self.grid.domain_shape_full(add=(-3, 0, 1)),
+        )
+        self.stencil_transport_flux = FixedOriginStencil(
+            transport_flux_xy,
+            origin=self.grid.compute_origin(),
+            domain=self.grid.domain_shape_compute(add=(1, 1, 1)),
```

Comment:
I know there has been a push to put the origin and domain in the init instead of in the call, this does help make sure you use the right axis offsets with your origin. It might make understanding the code in 'call' a bit harder, but I think is worth it for the speedup we can get from removing normalize_origin from __call__